src/plugins/jingle/tests/message-initiation.js

```diff
@@ -0,0 +1,58 @@
+/*global mock, converse */
+const u = converse.env.utils;
+const sizzle = converse.env.sizzle;
+
+const { Strophe } = converse.env;
+
+fdescribe("A Jingle Message Initiation Request", function () {
+
+    it("is sent when the user clicks the call button", mock.initConverse(
+        ['chatBoxesFetched'], {}, async function (_converse) {
+
+    await mock.waitForRoster(_converse, 'current', 1);
+    const contact_jid = mock.cur_names[0].replace(/ /g,'.').toLowerCase() + '@montague.lit';
+    await mock.openChatBoxFor(_converse, contact_jid);
+    const view = _converse.chatboxviews.get(contact_jid);
+    const call_button = view.querySelector('converse-jingle-toolbar-button button');
+    call_button.click();
+    const sent_stanzas = _converse.connection.sent_stanzas;
+    const stanza = await u.waitUntil(() => sent_stanzas.filter(s => sizzle(`propose[xmlns='${Strophe.NS.JINGLEMESSAGE}']`, s).length).pop());
+    expect(Strophe.serialize(stanza)).toBe(
+        `<message from='${_converse.jid}'
+            to='${contact_jid}'
+            type='chat'>`+
+                `<propose xmlns='${Strophe.NS.JINGLEMESSAGE}' id='${stanza.getAttribute('id')}'>`+
```

Comment:
`stanza.getAttribute('id')` will get the `id` attribute from the root of the stanza, which is the `message` element. You need to get it from the `propose` element.

Reply:
@jcbrand yeah about that, how do I get the `propose_id` object here https://github.com/conversejs/converse.js/pull/2956/files#diff-abe3b677d9605e0278786d222bb105d87006e60b51e0b531f6b82cd2e5cfa00cR34 ?

Reply:
The `propose` element is part of the `stanza` that you're testing here. So you can use `stanza.querySelector` to get the `propose` element and then get the id from it.

Reply:
> The `propose` element is part of the `stanza` that you're testing here. So you can use `stanza.querySelector` to get the `propose` element and then get the id from it.

that makes sense, thanks a lot